### ISSUE TITLE
any IPv6 network should stringify as ::/0

### ISSUE THREAD

**Issue by abraxxa** [maxmind/Net-Works#28]
currently ::0/0 which doesn't conform to the IPv6 standard.

**Comment by oschwald** [comment]
I agree this is weird, but it is not wrong. Both representations are valid. It looks like this was intentionally introduced in [this commit](https://github.com/maxmind/Net-Works/commit/6f3e842fa98a26e7a85096ee0e576db811784446).

**Comment by abraxxa** [comment]
@autarch can you please comment if there is a reason to stringify like this?

**Comment by autarch** [comment]
I really can't remember why this was changed. The commit that did this is from nearly seven years ago.

**Comment by abraxxa** [comment]
I finally found the RfC I was looking for: https://tools.ietf.org/html/rfc5952
Section 4.2 describes the shortening rules but doesn't explicitly mention this case.
All operating systems I've checked show :::* for sockets listening for any IPv6 address (the third : is the delimiter between address and port, really a great choice by the IPv6 designers to use the same as the delimiter between the fields).
Are there any objections against changing the stringification to ::/0?

**Comment by oschwald** [comment]
I am not opposed to a PR that does this and the RFC on the recommended formatting is a compelling argument. I would suspect that it was done as there was some variance between platforms on what `inet_ntop` would return.

**Comment by oschwald** [comment]
We are no longer actively maintaining this distribution. I am closing this issue as the repo will be archived.
